challenge-17/challenge-17.js

```diff
@@ -1,93 +1,123 @@
+	*/
+	console.log( '\nTrocando naturalidade:' );
+	console.log( text.replace( /brasileiro/, 'paulista' ) ); 
 
-/*
-Substitua todos os números por um traço `-`. Cada caractere de número deve
-ser um traço. Mostre o resultado no console:
-*/
-console.log( '\nTrocando números por -:' );
-// ?
+	/*
+	Substitua todos os números por um traço `-`. Cada caractere de número deve
+	ser um traço. Mostre o resultado no console:
+	*/
+	console.log( '\nTrocando números por -:' );
+	console.log( text.replace( /\d/g, '-' ) );
 
-/*
-Substitua todas as letras (somente letras) de "D" maiúsculo até "h"
-minúsculo por "0" (número zero). Mostre o resultado no console:
-*/
-console.log( '\nTrocando de "D" a "h" por "0":' );
-// ?
+	/*
+	Substitua todas as letras (somente letras) de "D" maiúsculo até "h"
+	minúsculo por "0" (número zero). Mostre o resultado no console:
+	*/
+	console.log( '\nTrocando de "D" a "h" por "0":' );
+	console.log( text.replace( /[D-Hd-h]/g, '0' ) );
 
-/*
-Substitua todos os "A" (maiúsculos ou minúsculos) por "4".
-Mostre o resultado no console:
-*/
-console.log( '\nTrocando "A" e "a" por "4":' );
-// ?
+	/*
+	Substitua todos os "A" (maiúsculos ou minúsculos) por "4".
+	Mostre o resultado no console:
+	*/
+	console.log( '\nTrocando "A" e "a" por "4":' );
+	console.log( text.replace( /A|a/g, '4' ) );
 
-/*
-Substitua a frase "O Centauro de Luvas", deixando-a em caixa alta, usando
-o método `toUpperCase()`. Mostre o resultado no console:
-*/
-console.log( '\n"O Centauro de Luvas" em caixa alta:' );
-// ?
+	/*
+	Substitua a frase "O Centauro de Luvas", deixando-a em caixa alta, usando
+	o método `toUpperCase()`. Mostre o resultado no console:
+	*/
+	console.log( '\n"O Centauro de Luvas" em caixa alta:' );
+	console.log( text.replace( /O Centauro de Luvas/g, function(frase) {
+		return frase.toUpperCase();
+	}) );
+	/*
+	Agora iremos substituir as datas no formato "13 de junho de 1804" para
+	"13/06/1804". A primeira coisa a fazer é criar uma função chamada
+	`getMonthNumber`, que irá receber um parâmetro (será o nome do mês) e deverá
+	retornar o número correspondente a esse mês.
+	Ex: Se o usuário entrar com "março", deve retornar "03" (em string mesmo).
+	Números com menos de dois dígitos devem ter um zero na frente.
+	Crie então a função e mostre no console os retornos para os meses de março,
+	setembro e dezembro.
+	Use um console.log para cada mês, usando a frase:
+	"O mês de [NOME DO MÊS] é representado pelo número [NÚMERO DO MÊS]."
+	*/
+	console.log( '\nMeses representados por números:' );
+	function getMonthNumber( monthName ) {
+ 		var months = {
+  			'janeiro': '1',
+  			'fevereiro': '2',
+  			'março': '3',
+  			'abril': '4',
+  			'maio': '5',
+  			'junho': '6',
+  			'julho': '7',
+  			'agosto': '8',
+  			'setembro': '9',
+  			'outubro': '10',
+  			'novembro': '11',
+  			'dezembro': '12'
+  		}
+  		return months[monthName] < 10 ? '0' + months[monthName] : months[monthName];
```

Comment:
-1
Já que está usando um objeto, e passando os números como string, já passe o zero direto no objeto. Se for retornar dessa forma, use um array e retorne o índice. Aí faz sentido validar se é menor que 10 :)